### PR TITLE
Add APIs for tree height

### PR DESCRIPTION
**Pull request by platfowner**
Summary:
- Add APIs `/get_state_info`, `ain_getStateInfo` for state info 
```
{
  tree_depth: <Tree depth>,
  tree_size: <Tree size>
}
``` 
- Rename state proof APIs: `/get_proof` -> `/get_state_proof`, `ain_getProof` -> `ain_getStateProof`
- Remove `GET_PROOF` operation as we have separate APIs 
- Tweak `getRefForWriting()` code

Testing:
- Passed unit tests and integration tests

Related issue:
- https://github.com/ainblockchain/ain-blockchain/issues/299

### CODE REVIEW

File: db/state-node.js [hunk 1]
```diff
@@ -13,21 +13,23 @@ class StateNode {
     // Used for leaf nodes only.
     this.value = null;
     this.proofHash = null;
+    this.treeDepth = 1;
```
Comment (liayoo) [inline db/state-node.js:16]: I think a depth of a tree is usually defined as "the length of the path to its root" ([ref](https://en.wikipedia.org/wiki/Tree_(data_structure)#Terminology)). Can we rename this to subtreeDepth or something else if you have a better name?

Comment (platfowner) [reply db/state-node.js]: Just replaced its name with 'tree height' and updated related logic accordingly. PTAL @liayoo 
Thanks for the clarification.

**Review by liayoo — APPROVED** [review]
LGTM, thanks!

**Comment by minsulee2** [comment]
LGTM tho, I've got a question that in what circumstance the state tree needs to have its height?

**Review by minsulee2 — APPROVED** [review]

**Comment by platfowner** [comment]
@minsulee2 The tree height is updated whenever db write happens (writeDatabase()) and will be used for setting limits on critical resources.

**Comment by platfowner** [comment]
Thanks for the reviews, now being merged..

**Pull request merged**